.github/workflows/run_test_windows.yml

```diff
@@ -0,0 +1,287 @@
+                      [boot]
+                      systemd=true
+                  set-as-default: 'true'
+                  wsl-shell-user: 'root'
+
+            - name: Upgrade to WSL 2
+              run: |
+                  echo "Current WSL status before upgrade:"
+                  wsl --status
+
+                  echo "Updating WSL kernel..."
+                  wsl --update --web-download
+
+                  echo "Setting default version to WSL 2..."
+                  wsl --set-default-version 2
+
+                  echo "Converting Ubuntu to WSL 2..."
+                  wsl --set-version Ubuntu 2
+
+                  echo "WSL status after upgrade:"
+                  wsl --status
+                  wsl --list --verbose
+              shell: powershell
+
+            - uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0
+
+            - name: Configure git for Unix line endings
+              run: |
+                  # Configure git to use LF line endings and fix any existing files
+                  git config core.autocrlf false
+                  git config core.eol lf
+                  # Reset and checkout to fix line endings
+                  git rm --cached -r .
+                  git reset --hard
+              shell: wsl-bash {0}
+
+            - uses: ./.github/actions/determine_workflow_context
+              id: set-conditions
+
+            - name: Print environment variables
+              run: |
+                  # Set workflow context variables inline
+                  AZLE_IS_MAIN_BRANCH_PUSH_FROM_FEATURE_MERGE="${{ steps.set-conditions.outputs.is_main_branch_push_from_feature_merge || 'false' }}"
+                  AZLE_IS_MAIN_BRANCH_PUSH_FROM_RELEASE_MERGE="${{ steps.set-conditions.outputs.is_main_branch_push_from_release_merge || 'false' }}"
+                  AZLE_IS_RELEASE_BRANCH_PR="${{ steps.set-conditions.outputs.is_release_branch_pr || 'false' }}"
+                  AZLE_IS_FEATURE_BRANCH_PR="${{ steps.set-conditions.outputs.is_feature_branch_pr || 'false' }}"
+                  AZLE_IS_FEATURE_BRANCH_DRAFT_PR="${{ steps.set-conditions.outputs.is_feature_branch_draft_pr || 'false' }}"
+
+                  echo "AZLE_IS_MAIN_BRANCH_PUSH_FROM_FEATURE_MERGE: $AZLE_IS_MAIN_BRANCH_PUSH_FROM_FEATURE_MERGE"
+                  echo "AZLE_IS_MAIN_BRANCH_PUSH_FROM_RELEASE_MERGE: $AZLE_IS_MAIN_BRANCH_PUSH_FROM_RELEASE_MERGE"
+                  echo "AZLE_IS_RELEASE_BRANCH_PR: $AZLE_IS_RELEASE_BRANCH_PR"
+                  echo "AZLE_IS_FEATURE_BRANCH_PR: $AZLE_IS_FEATURE_BRANCH_PR"
+                  echo "AZLE_IS_FEATURE_BRANCH_DRAFT_PR: $AZLE_IS_FEATURE_BRANCH_DRAFT_PR"
+              shell: wsl-bash {0}
+
+            - name: Report full path of test
+              # Just in case the path isn't obvious from the name, this will remove ambiguity
+              run: echo ${{matrix.test.path}}
+              shell: wsl-bash {0}
+
+            - name: Setup Node.js in WSL
```

Comment:
Let's discuss why the action doesn't work for this. If the action truly won't work, then I think we should use nvm directly. I don't see why nvm wouldn't work.

Reply:
got nvm to work, let's role with that for a bit and come back to using the action again after everything else has settled down